addons/project_timesheet_holidays/models/resource_calendar_leaves.py

```diff
@@ -143,6 +143,29 @@ def _timesheet_prepare_line_values(self, index, employee_id, work_hours_data, da
             'company_id': employee_id.company_id.id,
         }
 
+    def _generate_public_time_off_timesheets(self, employees):
+        timesheet_vals_list = []
+        work_hours_data = self._work_time_per_day()
+        for leave in self:
+            for employee in employees:
+                if employee.resource_calendar_id != leave.calendar_id:
+                    continue
+                work_hours_list = work_hours_data[leave.id]
+                for index, (day_date, work_hours_count) in enumerate(work_hours_list):
+                    existing_timesheets = leave["timesheet_ids"].filtered(
+                        lambda r: r.employee_id.id == employee.id and r.date == day_date)
+                    if existing_timesheets:
```

Comment:
In fact, we could also do a read_group before the loops to gather all timesheets per employee.

Something like:
```py
timesheet_read_group = self.env['account.analytic.line'].read_group(
    [('global_leave_id', 'in', self.ids), ('employee_id', 'in', employees.ids)], 
    ['date:array_agg'],
    ['employee_id'], 
)
timesheet_dates_per_employee_id = {
    res['employee_id'][0]: res['date'] 
    for res in timesheet_read_group
}
```

and so we could use it to know if there is a timesheet already generated in that day.

```suggestion
                timesheet_dates = timesheet_dates_per_employee_id.get(employee.id, [])
                for index, (day_date, work_hours_count) in enumerate(work_hours_list):
                    generate_timesheet = day_date not in timesheet_dates
                    if not generate_timesheet:
```

ps: I did not test my suggestion.